### ISSUE TITLE
Very eager configuration file watcher

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#147560]
Steps to Reproduce:

1. have vscode running already
2. from the command line run `code-insiders --verbose`
3. it should focus the running instance

From the output however it seems we setup a file watcher even though we terminate right after:

```
[main 2022-04-16T12:07:58.719Z] [File Watcher (node.js)] Request to start watching: /Users/bpasero/Library/Application Support/code-oss-dev/User (excludes: <none>, includes: <all>),/Users/bpasero/Library/Application Support/code-oss-dev/User/settings.json (excludes: <none>, includes: <all>)
[main 2022-04-16T12:07:58.728Z] Sending env to running instance...
[main 2022-04-16T12:07:58.734Z] [File Watcher (node.js)] Started watching: '/Users/bpasero/Library/Application Support/code-oss-dev/User/settings.json'
[main 2022-04-16T12:07:58.735Z] [File Watcher (node.js)] Started watching: '/Users/bpasero/Library/Application Support/code-oss-dev/User'
[main 2022-04-16T12:07:58.749Z] Sent env to running instance. Terminating...
[main 2022-04-16T12:07:58.749Z] Lifecycle#kill()
[main 2022-04-16T12:07:58.749Z] Lifecycle#onWillShutdown.fire()
```

I would suggest to install the listener later after we know we are the first instance if possible.

**Comment by sandy081** [comment]
[Initialising](https://github.com/microsoft/vscode/blob/8a39cb93047294ff10960f3313ac169aa2fcf5ac/src/vs/code/electron-main/main.ts#L98) of services happen before claiming the instance ie., checking for the first instance. Is it safe to move it after we claim the instance?

**Comment by bpasero** [comment]
Yeah that is a bit of a debt area: we actually have a set of services that are created all the time, even when a second instance is running and a set of services that we only create in the single instance. 

Maybe we should tackle this as part of adopting the `requestSingleInstanceLock` that @rzhao271 is working on: ideally services are not created unless we are truly running in the main instance. However, we have some code that needs to run even from the second instance, for example when you run `code --verbose` we still want the log service to work and that service may have dependencies to other services. Same goes for `code --status`...

**Comment by sandy081** [comment]
@bpasero Also assigned to you as I understood that this needs also assistance from you / @rzhao271 ?

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!

**Comment by bpasero** [comment]
I looked into this again and we could indeed move configuration service out of `main.ts` and into `app.ts` that is only being executed for the first instance. However, now I remember why we create config service in `main.ts`: we can `initialize` it in parallel with other things with initialize on startup:

https://github.com/microsoft/vscode/blob/04e34f25549ca272802d8532a5616b48ef262b4e/src/vs/code/electron-main/main.ts#L226-L246

Moving this to `app.ts` would result in some kind of waterfall because we need config service in backup service and that one also needs to be initialized:

https://github.com/microsoft/vscode/blob/d9373893583a1b78df346f4671bc24221d5e5aba/src/vs/code/electron-main/app.ts#L682

So we could not really run that in parallel as we can do in `main.ts`.

Lets leave it like that. The only possible fix I can see is to simply start the file watcher in `app.ts`, but not sure thats worth it...
